From 3180b6dd78d0defe93df48b25f6465ff3576c4ce Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Thu, 14 Mar 2019 13:41:46 +0100
Subject: [PATCH 2/4] Improve UiccSlot#promptForRestart dialog

* Disable dismissing dialog on outside touch and
  add negative button that will dismiss it instead.

Change-Id: Icdaad8cd0c328b5d1be1c137275f5ba899fd4f2f
---
 src/java/com/android/internal/telephony/uicc/UiccSlot.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/java/com/android/internal/telephony/uicc/UiccSlot.java b/src/java/com/android/internal/telephony/uicc/UiccSlot.java
index 25ef6376c..766f43c3c 100644
--- a/src/java/com/android/internal/telephony/uicc/UiccSlot.java
+++ b/src/java/com/android/internal/telephony/uicc/UiccSlot.java
@@ -306,13 +306,16 @@ public class UiccSlot extends Handler {
             String message = (isAdded) ? r.getString(R.string.sim_added_message) :
                     r.getString(R.string.sim_removed_message);
             String buttonTxt = r.getString(R.string.sim_restart_button);
+            String buttonCancelTxt = r.getString(R.string.cancel);
 
             AlertDialog dialog = new AlertDialog.Builder(mContext)
                     .setTitle(title)
                     .setMessage(message)
                     .setPositiveButton(buttonTxt, listener)
+                    .setNegativeButton(buttonCancelTxt, null)
                     .create();
             dialog.getWindow().setType(WindowManager.LayoutParams.TYPE_SYSTEM_ALERT);
+            dialog.setCanceledOnTouchOutside(false);
             dialog.show();
         }
     }
-- 
2.17.1

